numpyro/infer/reparam.py

```diff
@@ -132,14 +132,13 @@ class TransformReparam(Reparam):
     def __call__(self, name, fn, obs):
         assert obs is None, "TransformReparam does not support observe statements"
         fn, expand_shape, event_dim = self._unwrap(fn)
-        if isinstance(fn, (dist.Uniform, dist.TruncatedCauchy, dist.TruncatedNormal)):
+        if not isinstance(fn, dist.TransformedDistribution):
             raise ValueError(
                 "TransformReparam does not automatically work with {}"
                 " distribution anymore. Please explicitly using"
                 " TransformedDistribution(base_dist, AffineTransform(...)) pattern"
                 " with TransformReparam.".format(type(fn).__name__)
             )
-        assert isinstance(fn, dist.TransformedDistribution)
```

Comment:
couple of comments here:
- the `assert` wouldn't run when executing Python in optimized mode
- as `dist.TruncatedNormal` is a function, `isinstance(fn, dist.TruncatedNormal)` isn't allowed

Any objections to just raising if `fn` isn't an instance of `dist.TransformedDistribution`?

Reply:
Those are good points! Thanks for making this enhancement!